### PR TITLE
story 11496 (collect|archive-search): update 10k items management

### PR DESCRIPTION
**Pull request by Regzox**
## Description

* Fix le retour API pour gérer les 10k+ items côté front
* Commonise la gestion des 10k+ items (collect|archive-search)
* Fix de traductions
* Fix incohérence sélection 10k+
* Suppression de fichier inutiles

### CODE REVIEW

**Comment by TDevillechabrolle** [comment]
Logo
**Checkmarx One – Scan Summary & Details** – [1af1c400\-414d\-48ad\-a874\-e61a5a09c417](https://eu.ast.checkmarx.net/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/scans?id=1af1c400-414d-48ad-a874-e61a5a09c417&branch=story_11496_track_total_hits)
### New Issues
Severity|Issue|Source File / Package|Checkmarx Insight
---|---|---|---
MEDIUM|[CVE-2016-10735](https://devhub.checkmarx.com/cve-details/CVE-2016-10735)|Npm\-bootstrap\-3\.3\.6|[Vulnerable Package](https://eu.ast.checkmarx.net/sca/#/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/reports/1af1c400-414d-48ad-a874-e61a5a09c417/vulnerabilities/CVE-2016-10735%3ANpm-bootstrap-3.3.6/vulnerabilityDetails)
MEDIUM|[CVE-2018-14040](https://devhub.checkmarx.com/cve-details/CVE-2018-14040)|Npm\-bootstrap\-3\.3\.6|[Vulnerable Package](https://eu.ast.checkmarx.net/sca/#/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/reports/1af1c400-414d-48ad-a874-e61a5a09c417/vulnerabilities/CVE-2018-14040%3ANpm-bootstrap-3.3.6/vulnerabilityDetails)
MEDIUM|[CVE-2018-14042](https://devhub.checkmarx.com/cve-details/CVE-2018-14042)|Npm\-bootstrap\-3\.3\.6|[Vulnerable Package](https://eu.ast.checkmarx.net/sca/#/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/reports/1af1c400-414d-48ad-a874-e61a5a09c417/vulnerabilities/CVE-2018-14042%3ANpm-bootstrap-3.3.6/vulnerabilityDetails)
MEDIUM|[CVE-2018-20676](https://devhub.checkmarx.com/cve-details/CVE-2018-20676)|Npm\-bootstrap\-3\.3\.6|[Vulnerable Package](https://eu.ast.checkmarx.net/sca/#/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/reports/1af1c400-414d-48ad-a874-e61a5a09c417/vulnerabilities/CVE-2018-20676%3ANpm-bootstrap-3.3.6/vulnerabilityDetails)
MEDIUM|[CVE-2018-20677](https://devhub.checkmarx.com/cve-details/CVE-2018-20677)|Npm\-bootstrap\-3\.3\.6|[Vulnerable Package](https://eu.ast.checkmarx.net/sca/#/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/reports/1af1c400-414d-48ad-a874-e61a5a09c417/vulnerabilities/CVE-2018-20677%3ANpm-bootstrap-3.3.6/vulnerabilityDetails)
MEDIUM|[CVE-2019-8331](https://devhub.checkmarx.com/cve-details/CVE-2019-8331)|Npm\-bootstrap\-3\.3\.6|[Vulnerable Package](https://eu.ast.checkmarx.net/sca/#/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/reports/1af1c400-414d-48ad-a874-e61a5a09c417/vulnerabilities/CVE-2019-8331%3ANpm-bootstrap-3.3.6/vulnerabilityDetails)
MEDIUM|[Cxf0b588a3-5c6f](https://devhub.checkmarx.com/cve-details/Cxf0b588a3-5c6f)|Npm\-jquery\-2\.2\.4|[Vulnerable Package](https://eu.ast.checkmarx.net/sca/#/projects/78091106-f137-4b7b-8b68-77ee0f4d3433/reports/1af1c400-414d-48ad-a874-e61a5a09c417/vulnerabilities/Cxf0b588a3-5c6f%3ANpm-jquery-2.2.4/vulnerabilityDetails)

### Fixed Issues
Severity|Issue|Source File / Package|Checkmarx Insight
---|---|---|---
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [150](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L150)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=kscBSNfsLifnc1Kq3YpGrN10pks%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [110](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L110)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=aUZHycZS4msMZ7wz80hQsi2aeIk%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [118](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L118)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=SfY8pRwd4Uv%2BFoLS%2B0693atnVuo%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [150](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L150)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=ay9hBkwgQ9ujzKXfZYRiyE73hdU%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [110](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L110)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=ekiu2nVzGUJCr75v%2B5Y%2Fy9RFJ5o%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [118](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L118)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=n2rBo%2BeIKZ0L1J9iytlku4tbMlo%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [150](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L150)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=EL9dChMQAy9V7%2BqEp4U06LYvAdo%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [110](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L110)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=4im%2Bo8x4572aCzgExsd8fPak5Z8%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [118](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L118)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=1ZEpkwCGwFEkIX9fHAAXn%2F%2B1gC8%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [150](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L150)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=VJiqSAzXMN7C%2BTdtpsMMR88YyW4%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [110](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L110)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=a8B9U8DD9QCNL6oT9kSUozbAMsE%3D&redirect=true)
MEDIUM|[Privacy_Violation](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/359/12956636075206043460)|/api/api\-iam/iam\-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService\.java: [118](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-iam/iam-security/src/main/java/fr/gouv/vitamui/iam/security/service/InternalSecurityService.java#L118)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=SX2jXxGs4a5lUTCmovc6FyioulY%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/OntologyExternalController\.java: [118](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/OntologyExternalController.java#L118)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=4Nqgcpn2PPWqhW1fLq5AfIdfeKU%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-collect/collect\-internal/src/main/java/fr/gouv/vitamui/collect/internal/server/rest/TransactionArchiveUnitInternalController\.java: [162](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-collect/collect-internal/src/main/java/fr/gouv/vitamui/collect/internal/server/rest/TransactionArchiveUnitInternalController.java#L162)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=h3eM8c63kajP0qWmtFofe7cFlxU%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/ui/ui\-identity/src/main/java/fr/gouv/vitamui/identity/rest/OwnerController\.java: [94](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//ui/ui-identity/src/main/java/fr/gouv/vitamui/identity/rest/OwnerController.java#L94)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=bng1rl4YbVBKXoAk9dUeST2DAv0%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/RuleExternalController\.java: [123](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/RuleExternalController.java#L123)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=PwKDrN2tnkcYxXbtv2f0KabCgTA%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/ui/ui\-identity/src/main/java/fr/gouv/vitamui/identity/rest/UserInfoController\.java: [86](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//ui/ui-identity/src/main/java/fr/gouv/vitamui/identity/rest/UserInfoController.java#L86)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=q%2FqXExzRjRblDHzFZjy7SEQkvxo%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ProfileExternalController\.java: [122](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ProfileExternalController.java#L122)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=Osyb8GE%2BvscvQuFey0EGy4Ph8ew%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ContextExternalController\.java: [106](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ContextExternalController.java#L106)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=0nY%2F2tnw%2Fy1UoLC1ieuT2vwtd%2FI%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/IngestContractExternalController\.java: [107](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/IngestContractExternalController.java#L107)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=0gUbu1QalDIYamGOk8TT0mXZcgY%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/ui/ui\-ingest/src/main/java/fr/gouv/vitamui/ingest/rest/IngestController\.java: [118](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//ui/ui-ingest/src/main/java/fr/gouv/vitamui/ingest/rest/IngestController.java#L118)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=ZI7Z%2Flc6zbGT%2F5190Se9%2FB7hrRg%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-collect/collect\-external/src/main/java/fr/gouv/vitamui/collect/external/server/rest/ProjectExternalController\.java: [186](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-collect/collect-external/src/main/java/fr/gouv/vitamui/collect/external/server/rest/ProjectExternalController.java#L186)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=VWpQN0E194Cwan3fPT7MxkNjvlQ%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-ingest/ingest\-external/src/main/java/fr/gouv/vitamui/ingest/external/server/rest/IngestExternalController\.java: [117](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-ingest/ingest-external/src/main/java/fr/gouv/vitamui/ingest/external/server/rest/IngestExternalController.java#L117)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=rS0blZFxe1dXtEkY7ZJaRQXWlLQ%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/ui/ui\-identity/src/main/java/fr/gouv/vitamui/identity/rest/UserController\.java: [95](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//ui/ui-identity/src/main/java/fr/gouv/vitamui/identity/rest/UserController.java#L95)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=aEUcsuPAyAdWlZux9yUtvv4iE8A%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ArchivalProfileUnitExternalController\.java: [117](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ArchivalProfileUnitExternalController.java#L117)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=XlWo2gqhGBrPsuyHAZqW244VsnQ%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/AgencyExternalController\.java: [115](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/AgencyExternalController.java#L115)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=FTzmZ1BT2b%2B1q336xyTbbvwCV0M%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ManagementContractExternalController\.java: [113](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/ManagementContractExternalController.java#L113)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=xYWpkbQusH%2F4onJTVn3uR9lpkJg%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/SecurityProfileExternalController\.java: [109](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/SecurityProfileExternalController.java#L109)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=D9AvUzLizOWaA0X9S9yQinuE7Z8%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/ui/ui\-identity/src/main/java/fr/gouv/vitamui/identity/rest/CustomerController\.java: [124](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//ui/ui-identity/src/main/java/fr/gouv/vitamui/identity/rest/CustomerController.java#L124)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=3SQNzrMkJ9hyyyPzbAXiY1Z%2FvAE%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/ui/ui\-collect/src/main/java/fr/gouv/vitamui/collect/rest/ProjectController\.java: [180](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//ui/ui-collect/src/main/java/fr/gouv/vitamui/collect/rest/ProjectController.java#L180)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=Vyo%2F8GRu4k9OoJWvXNQVzRKzo4I%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/api/api\-referential/referential\-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/AccessContractExternalController\.java: [104](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//api/api-referential/referential-external/src/main/java/fr/gouv/vitamui/referential/external/server/rest/AccessContractExternalController.java#L104)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=fRg0i1Tdccvfe4ngVx744rt3%2BUI%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/ui/ui\-commons/src/main/java/fr/gouv/vitamui/ui/commons/rest/ExternalParamProfileController\.java: [97](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//ui/ui-commons/src/main/java/fr/gouv/vitamui/ui/commons/rest/ExternalParamProfileController.java#L97)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=jN4AlGjaVR%2BeM%2BELlwrHRdFHoS4%3D&redirect=true)
LOW|[Log_Forging](https://eu.ast.checkmarx.net/results/78091106-f137-4b7b-8b68-77ee0f4d3433/1af1c400-414d-48ad-a874-e61a5a09c417/sast/description/117/9884244256835554923)|/ui/ui\-identity/src/main/java/fr/gouv/vitamui/identity/rest/TenantController\.java: [91](https://github.com/ProgrammeVitam/vitam-ui/blob/story_11496_track_total_hits//ui/ui-identity/src/main/java/fr/gouv/vitamui/identity/rest/TenantController.java#L91)|[Attack Vector](https://eu.ast.checkmarx.net/results/1af1c400-414d-48ad-a874-e61a5a09c417/78091106-f137-4b7b-8b68-77ee0f4d3433/sast?result-id=DG9IEmEURfPeqDdEFFJE8D1JNO4%3D&redirect=true)

**Review by laedanrex — APPROVED** [review]

File: ui/ui-frontend-common/src/app/modules/archive/components/archive-unit-count/archive-unit-count.component.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      this.pending = value;
+      this.pendingChange.emit(this.pending);
+    };
+    const notPending = () => {
+      updatePending(false);
+    };
+
+    updatePending(true);
+    this.subscriptions.add(
+      this.search.pipe(tap(notPending, notPending)).subscribe(
```
Comment (laedanrex) [inline ui/ui-frontend-common/src/app/modules/archive/components/archive-unit-count/archive-unit-count.component.ts:104]: tu voulais en mettre 1 seul je pense     
et dans ce cas que pense tu de direct appeler la fonction genre tap(()-> updatePending(false)) ?

Comment (laedanrex) [reply ui/ui-frontend-common/src/app/modules/archive/components/archive-unit-count/archive-unit-count.component.ts:104]: au passage ne connaissant pas bien tap je suis aller regardé et ca semble deprecié dans les versions plus hautes d'angular.

Comment (Regzox) [reply ui/ui-frontend-common/src/app/modules/archive/components/archive-unit-count/archive-unit-count.component.ts:104]: tap(notPending, notPending)

* premier pour le onSuccess
* second pour le onError

Je cherche à reset le pending quoiqu'il arrive dans cette étape.

**Review by mhameg — APPROVED** [review]

**Pull request merged**